### PR TITLE
I924 parquet io

### LINKED ISSUE #924: Create SmvParquetIoStrategy to persist intermediate results

### PR DESCRIPTION
**Pull request by ninjapapa**
Fixed #924 

Created `SmvParquetOnHdfsIoStrategy`. Will create another issue to switch the default persist to parquet.

### CODE REVIEW

File: src/main/python/smv/smviostrategy.py [hunk 1]
```diff
@@ -38,20 +38,18 @@ def isPersisted(self):
     def remove(self):
         """Remove persisted file(s)"""
 
-# TODO: add lock, add publish
-class SmvCsvOnHdfsIoStrategy(SmvIoStrategy):
-    def __init__(self, smvApp, fqn, ver_hex, csv_path=None):
+
+class SmvFileOnHdfsIoStrategy(SmvIoStrategy):
```
Comment (AliTajeldin) [inline src/main/python/smv/smviostrategy.py]: need a doc do describe this base class and what is the derived classes supposed to overwrite/provide.

File: src/main/python/smv/smviostrategy.py [hunk 2]
```diff
@@ -60,32 +58,53 @@ def _smvLock(self):
...
+    def remove(self):
+        self.smvApp._jvm.SmvHDFS.deleteFile(self._file_path)
+
+
+class SmvCsvOnHdfsIoStrategy(SmvFileOnHdfsIoStrategy):
+    def __init__(self, smvApp, fqn, ver_hex, file_path=None):
+        super(SmvCsvOnHdfsIoStrategy, self).__init__(smvApp, fqn, ver_hex, 'csv', file_path)
+
+    @property
+    def _schema_path(self):
```
Comment (AliTajeldin) [inline src/main/python/smv/smviostrategy.py]: Where is this file being written?  I can see it being deleted, but not written.

File: src/main/python/smv/smviostrategy.py [hunk 3]
```diff
@@ -94,22 +113,27 @@ def isPersisted(self):
...
-        return self._jvm.SmvHDFS.exists(self.path)
 
-    def remove(self):
-        self._jvm.SmvHDFS.deleteFile(self.path)
\ No newline at end of file
+class SmvParquetOnHdfsIoStrategy(SmvFileOnHdfsIoStrategy):
+    def __init__(self, smvApp, fqn, ver_hex, file_path=None):
+        super(SmvParquetOnHdfsIoStrategy, self).__init__(smvApp, fqn, ver_hex, 'parquet', file_path)
+
+    def read(self):
```
Comment (AliTajeldin) [inline src/main/python/smv/smviostrategy.py]: There seems to be a mismatch between the read/write names.  We should be consistent with the names of the functions that the derived classes would overwrite.  Either both start with `_` or both without.

File: src/main/python/smv/smviostrategy.py (hunk 1 above)
Comment (ninjapapa) [reply src/main/python/smv/smviostrategy.py]: will do.

File: src/main/python/smv/smviostrategy.py (hunk 2 above)
Comment (ninjapapa) [reply src/main/python/smv/smviostrategy.py]: Scala side call below wrote 2 files, csv and the schema file:
`self.smvApp.j_smvPyClient.persistDF(self._file_path, jdf)`

File: src/main/python/smv/smviostrategy.py (hunk 3 above)
Comment (ninjapapa) [reply src/main/python/smv/smviostrategy.py]: will do.

**Comment by ninjapapa** [comment]
Changes made as required.

**Comment by ninjapapa** [comment]
@AliTajeldin ready to go?

**Pull request merged**